### ISSUE TITLE
"Immutable workspace contents have been modified" failure when workspace contents are not modified

### ISSUE THREAD

**Issue by bigdaz** [gradle/gradle#28475]
### Current Behavior

A build can fail with `Immutable workspace contents have been modified` without any modification of the directory in question.

### Expected Behavior

This error should only occur when files in the workspace have been modified directly or by a non-Gradle process.

### Context (optional)

A few users have reported having builds fail with `Immutable workspace contents have been modified` on CI, where it is very unlikely that these workspace files have been directly modified. Instead, it appears that some scenario driven by Gradle can produce this error.

I managed to reproduce this scenario once using a [fork of the repository](https://github.com/bigdaz/Anki-Android) where this [issue was reported to gradle/actions](https://github.com/gradle/actions/issues/47).

[Here is the workflow run demonstrating this failure](https://github.com/bigdaz/Anki-Android/actions/runs/8251951910/job/22570285686#step:10:155). 

I have captured the offending `transforms-4` directory immediately prior to the build failing, which allows offline inspection of the contents. However, when I inspect the `metadata.bin` file and calculate the output hashes (using [logic copied from AssignImmutableWorkspaceStep](https://github.com/gradle/gradle/blob/c0cd6d646441a195bb0ba2d9762b3130dd96fd4a/platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/AssignImmutableWorkspaceStep.java#L108-L113)), then the hashes in `metadata.bin` exactly match those calculated.

I don't have an explanation for why the error is being raised for this particular `transforms-4` workspace directory, but it caused the build to fail consistently and repeatedly once in place.

Here's the data extracted from the `transforms-4/f46cf63d0a549c8bc814e25133f449eb` workspace:
```
Metadata for f46cf63d0a549c8bc814e25133f449eb
OriginMetadata{buildInvocationId=v5utnjtubnbupherwskzgpwdxy, executionTime=PT1.468S}
{outputDirectory=[33a5350884b175260845f1ad15ac99cf], resultsFile=[91d756dd3d78b7213fe867e6756372d4]}
-----
Calculated output hashes
{outputDirectory=[33a5350884b175260845f1ad15ac99cf], resultsFile=[91d756dd3d78b7213fe867e6756372d4]}
```

### Steps to Reproduce

This error is difficult to reproduce, and I have only managed to do so once. However, once the error first appears, it is persistent until the underlying `transforms-4/XX` directory is removed.

### Gradle version

8.6

**Comment by bigdaz** [comment]
@gradle/bt-execution a thought: since the actual workspace directory appears to be internally consistent in the failing case, is it possible that we are incorrectly using a previously calculated value for the hash of `transformed/android.jar`, rather than recalculating the hash of the actual contents? 

[Looking at the code](https://github.com/gradle/gradle/blob/c0cd6d646441a195bb0ba2d9762b3130dd96fd4a/platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/AssignImmutableWorkspaceStep.java#L108-L111) it appears the hash is being recalculated, but I'm not sure what caching might be in place.

**Comment by lptr** [comment]
Here's a theory that seems to fit what we see. Consider the following chain of events:

1) A build is executed that triggers the execution of our transform in question, and output contents "version 1" is produced in the `$GUH/caches/transforms-4/XXX` workspace directory.
2) Something causes `transforms-4/XXX` to be cleaned up (but not all of the Gradle user home (GUH) caches). This can be the Gradle GitHub Action aggressively deleting entries untouched during the build (say, after a build that does _not_ trigger the transform), or the regular GUH cleanup triggering, or some sort of other cleanup routine.
3) Another build is executed, which correctly recognizes that the `transforms-4/XXX` workspace directory is missing, and proceeds to recreate it. It's important that it produces byte-for-byte _different_ output, such as a JAR with different timestamps.
4) Yet another build is executed that once again triggers the transform. Gradle recognizes that `transforms-4/XXX` is available, and proceeds to reuse it, but then the validation of the content on disk fails.

The culprit could be the `CachingFileHasher` used for caching file hashes during and between builds in the GUH caches. This cache is persisted to disk, so it survives even daemon restarts, and the GitHub Action copies it (along with the rest of the global caches) to every executor. So even ephemeral GitHub builds get a warm file hash cache.

The assumption is that this file hash cache retains hashes for the "version 1" output:

a) when it's deleted from disk, and 
b) then when it is subsequently recreated as "version 2", the old content is not recognized as stale.

If these hold true, then build 4) above would read the "version 2" of `metadata.bin` from disk with the hash for the "version 2" outputs, but when it snapshotted the outputs, it would get stale hashes from the file hash cache.

We know that a) is true because `LeastRecentlyUsedCacheCleanup` doesn't remove entries from the file hash cache when it deletes cache content.

As for b), with the GitHub Action we know that the action itself tampers with file timestamps in the global caches. It's a bit unclear whether in the non-ephemeral case we could also somehow end up with `GradleUserHomeScopeFileTimeStampInspector` not recognizing a file in the global caches as changed, and returning the old timestamp.

## TODO

1) Add a flag to disable the consistency check to unblock the Gradle 8.7 release
2) Add some more diagnostics to confirm assumptions about stale content in the file hash cache being the root cause of our problems.
3) Either stop the Gradle GitHub Action from tampering with file timestamps in the global caches, or exclude the file hash cache files from the caches it carries from executor to executor.
3) Let `LeastRecentlyUsedCacheCleanup` remove deleted entries from the file hash cache. It should also invalidate any relevant parts of the VFS. (As a simplification we could just drop both caches whenever a cleanup happens.)
4) Integrate the file hash cache and the VFS more closely; make sure the relevant entries are removed from the file hash cache when a part of the VFS is invalidated. We should do this with `FileContentCache` as well. There should be a single mechanism inside a daemon to invalidate everything we know about a location, and it should invalidate all caches.
5) Implement a mechanism where cache cleanup can force other daemons to drop deleted content from both the file hash cache and the VFS. It would be ideal to make sure no other daemon actually runs while cache cleanup happens.

**Comment by lptr** [comment]
Actually, for the non-ephemeral case, when we have the daemon surviving between builds, it's possible the VFS itself retains the stale hashes after the cleanup. FTR, the services wrap each other thusly:

```mermaid
graph TD
  AssignImmutableWorkspaceStep:::highlight --> FileCollectionSnapshotter
  FileCollectionSnapshotter --> OutputSnapshotter
  OutputSnapshotter --> FileSystemAccess
  FileSystemAccess --> VirtualFileSystem
  VirtualFileSystem:::highlight --> FileHasher:::highlight

  classDef highlight stroke-width:4px
```

**Comment by lamphamTL** [comment]
I consistently get this issue with the task `DexMergingTask`, with or without daemon, both on CI and locally

**Comment by lamphamTL** [comment]
> I consistently get this issue with the task DexMergingTask, with or without daemon, both on CI and locally.
I'm using Gradle 8.6

To add a bit more context about this, we are trying to upgrade our gradle from 7.6.4 to 8.6, which means on CI there are mixed cache between 8.6 (for jobs on the upgrade branch) and 7.6.4 (for jobs on other branches).
Same in local, caches are mixed between 7.6.4 and 8.6.

When I removed GUH caches in local, the problem seems (it was consistent failing, and now no build so far fails) to disappear.
Didn't try with CI because it requires some further setups (removing caches on a CI agent and preventing other branches, which are based on 7.6.4, from getting executed on this agent)

**Comment by asodja** [comment]
@lamphamTL can you maybe create a reproducer and share it with us? That would help us a lot. 
More information about your setup could also help: e.g. Operating system, filesystem, AGP version, task you run and where it fails etc.

**Comment by bigdaz** [comment]
@lamphamTL when you say "consistently", does this mean that the issue reoccurs even after you remove the offending `transforms-4/XXX` directory? If not, please make a full copy of your `~/.gradle` directory while your project is in a bad state.

Then, some experiments to try:
1) Run the build with a clean Gradle User Home (ie `./gradlew -g HOME ...`
2) Run the build with a fresh clone of your project repository

Knowing if these experiments can reproduce the issue will help. We are having trouble getting a reliable reproducer for this issue, so any evidence you can provide may be helpful.

**Comment by lamphamTL** [comment]
@bigdaz  no, removing GUH caches did make the first build to work; however, it starts failing consistently from the second one

**Comment by lptr** [comment]
I've made the handling of the problem more sophisticated in:

- https://github.com/gradle/gradle/pull/28503

With this change when Gradle bumps into an inconsistent workspace, it will not fail the build, but instead discard the workspace and re-run the work. It will also produce a deprecation warning with some details:

```text
> Transform lib1.jar (project :lib) with Duplicator
The contents of the immutable workspace '...' have been modified. This behavior has been deprecated. This will fail with an error in Gradle 9.0. These workspace directories are not supposed to be modified once they are created. The modification might have been caused by an external process, or could be the result of disk corruption. The inconsistent workspace has been moved to '...', and will be recreated.
outputDirectory:
 - transformed (Directory, 7cfc7dbc0901a39cf9a5f44f7ea491b8)
   - 0 (Directory, 78af062fd74d95b8d382f5adabb136f5)
     - lib1-green.jar (RegularFile, 1ba4170a71868b572bca2e747af7a707)
   - tamper-tamper.txt (RegularFile, c2a9b87422fcb39f4f55ff182939992b)

resultsFile:
 - results.bin (RegularFile, 44ec6391aaade3532e59b1d99d292b4d)
```

The more sophisticated handling of the problem will be included in 8.7-rc-4 (soon to be released).

I moved this issue to 8.8-rc-1, as we still need to figure out what causes the false positives, and come up with a fix.

**Comment by mustafa-antonio** [comment]
@lptr I encountered this error today on my GitLab CI server after update to Gradle 8.6. I'm using remote Gradle cache to share between branches therefore Gradle cache directory changes every time. 

It happens for all branches which fetched cache created by `main` branch so maybe that's a hint to reproduce it.

Also not sure if relevant but I'm also using custom `GRADLE_ENCRYPTION_KEY` on my CI.

**Comment by lptr** [comment]
Could you please try using 8.7-rc-4? It should have the improved behavior where we don't fail the build anymore.

**Comment by lptr** [comment]
@LukasVykuka that is a different problem. I suspect this is caused by a virus scanner checking your Gradle user home, and keeping files open, preventing Gradle from moving the directory containing them.

**Comment by LukasVykuka** [comment]
@lptr good point but I have Gradle home and project folders excluded from antivirus scanning. But it turns out that it is issue with Adnroid studio and "Sync with Gradle" function. Building from command line it work without issue.

**Comment by mustafa-antonio** [comment]
@lptr FYI Gradle 8.7 fixed my problems 🚀

**Comment by vickyleu** [comment]
> @lptr仅供参考 Gradle 8.7 解决了我的问题 🚀

solved my problem

**Comment by neeldoshii** [comment]
To those who are facing this issue try upgrading the gradle distribution url to 8.7
```distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip```

**Comment by dodoto** [comment]
> To those who are facing this issue try upgrading the gradle distribution url to 8.7对于那些面临此问题的人，请尝试将 gradle 分发 url 升级到 8.7 `distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip`

I upgraded url to 8.7. Unfortunally still get the error: java.io.UncheckedIOException: Could not move temporary workspace.

**Comment by allenpbiju** [comment]
`distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip`

I also upgraded to 8.7 and I am still getting the error

`java.io.UncheckedIOException: Could not move temporary workspace (D:\dev\RNExperiments\android\.gradle\8.7\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-8a203d1d-3ff5-4f9d-ad38-d7188fa07baa) to immutable location (D:\dev\RNExperiments\android\.gradle\8.7\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a)`

OS Version : Windows 11

**Comment by saulonunesdev** [comment]
i just tested on a fresh windows 11 machine with AwesomeProject, and the same error appears for me

**Comment by priyanshu20022** [comment]
> `distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip`
> 
> I also upgraded to 8.7 and I am still getting the error
> 
> `java.io.UncheckedIOException: Could not move temporary workspace (D:\dev\RNExperiments\android\.gradle\8.7\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-8a203d1d-3ff5-4f9d-ad38-d7188fa07baa) to immutable location (D:\dev\RNExperiments\android\.gradle\8.7\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a)`
> 
> OS Version : Windows 11

Hi Allen, I am facing the same issue. Have you found any solution?

**Comment by allenpbiju** [comment]
I ran the command `npm run android` seperately and the build gets successful.

**Comment by priyanshu20022** [comment]
> I ran the command `npm run android` seperately and the build gets successful.
@allenpbiju Thank you sir 🥺🥺🙏

**Comment by saulonunesdev** [comment]
updating

`distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip`

in `android/gradle/wrapper/gradle-wrapper.properties`

Worked for me

**Comment by yelnarshopanov** [comment]
> > @LukasVykuka that is a different problem. I suspect this is caused by a virus scanner checking your Gradle user home, and keeping files open, preventing Gradle from moving the directory containing them.
> 
> Thx! That really helps. After uninstalling the RAV Antivirus that got installed on the computer at some point, everything worked fine. Was using Gradle 8.7.

uninstalling the RAV Antivirus, Worked for me

**Comment by benardelia** [comment]
> To those who are facing this issue try upgrading the gradle distribution url to 8.7 `distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip`

this work fine for me

**Comment by featlast** [comment]
Greetings, any solution, for the issue in question, I made this manual gradle configuration and it still doesn't work 
![image](https://github.com/gradle/gradle/assets/66761042/6075bfef-a8ef-4d53-a988-d27ff0fc188f)

**Comment by benardelia** [comment]
Change cradle distribution url direct to the 

Your project > gradle > wrapper > gradle-wrapper.properties

**Comment by benardelia** [comment]
Change cradle distribution url direct to the

Your project > gradle > wrapper > gradle-wrapper.properties

**Comment by featlast** [comment]
Greetings @benardelia , I already made the change to different versions and it still gives me the same error

> Cambie la URL de distribución de la cuna directamente al
> 
> Su proyecto > gradle > contenedor > gradle-wrapper.properties

**Comment by lptr** [comment]
@featlast do you have a virus scanner that might be reading the Gradle user home caches in `$USER_HOME\.gradle\caches`?

**Comment by benardelia** [comment]
> Greetings @benardelia , I already made the change to different versions and it still gives me the same error
> 
> > Cambie la URL de distribución de la cuna directamente al
> > Su proyecto > gradle > contenedor > gradle-wrapper.properties

This occurred when a specific file modified or even deleted manually.  please can you share the full logs

**Comment by featlast** [comment]
Hello @lptr , I was carrying out the analysis with the Windows Defender antivirus and everything came out ok; Hello @benardelia  as for the configuration sometimes, when you close and open the project again it allows the gradle to synchronize, sometimes it works well, other times it doesn't, open it and wait for a patch by gradle to solve this which becomes very uncomfortable when creating a new project.

**Comment by benardelia** [comment]
Thanks for the feedback.

**Comment by undefine-man** [comment]
Face problem
`java.io.UncheckedIOException: Could not move temporary workspace (C:\Users\xx\Desktop\workspace\cc\android\.gradle\8.8\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-53c65751-a005-4330-bd6a-3da56fd2bc22) to immutable location (C:\Users\xx\Desktop\workspace\cc\android\.gradle\8.8\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a)
> Could not move temporary workspace (C:\Users\xx\Desktop\workspace\cc\android\.gradle\8.8\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-53c65751-a005-4330-bd6a-3da56fd2bc22) to immutable location (C:\Users\xx\Desktop\workspace\cc\android\.gradle\8.8\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a)`
when start the react native project.

react native use gradle 8.7 as default now. 
I tried disabling Microsoft Defender, but the problem remians.
Only downgrade to gradle `8.5` solved the problem. Gradle version above `8.5` will lead to the problem

Problem solved by changing
 `distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip`  to `distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip` 
in file `gradle-wrapper.properties` .

**Comment by raamcosta** [comment]
In hopes of helping someone going through this, without reverting to old gradlew version, nuking all caches worked for me (obviously I suppose 😛), just make sure to stop gradle daemons first:

`./gradlew --stop`
`rm -rf ~/.gradle/caches/ `

**Comment by jalupriyangga** [comment]
I use mcafee antivirus regularly, and facing the same problem. Everytime I make new project, I turn off the antivirus scanner and it worked for me

**Comment by Harshgour2002** [comment]
Upgrade to gradle 8.7 Don't works for me, still facing this issue with the transform4 folder

**Comment by asodja** [comment]
Do you experience that a problem also with Gradle 8.8?

**Comment by featlast** [comment]
Greetings, some solution to this problem, I have not been able to create projects in jetpackcompose again, I cannot work. Aid...😖

**Comment by cblaze22** [comment]
This is also happening to me after upgrading to latest React Native. java.lang.IllegalStateException: Immutable workspace contents have been modified:  

I have to delete the .gradle version for it to build properly.

**Comment by davla86** [comment]
I solved this problem after deleting RAV endpoint.

**Comment by MateoLa** [comment]
Some coments here:
  -  I'm not admin of my windows pc so I coudn't stop the antivirus when I run >npm run start 
  -  I was resolving this issue by downgrading to gradle 8.5 in the distributionUrl. Now with the latest react-native (0.75.1) I am getting the following error message: 

```
> Failed to apply plugin 'com.android.internal.version-check'.
   > Minimum supported Gradle version is 8.7. Current version is 8.5. If using the gradle wrapper, try editing the distributionUrl in C:\Users\c510268\Desktop\React-Native\07-componentsApp\android\gradle\wrapper\gradle-wrapper.properties to gradle-8.7-all.zip
```

**Comment by boiboif** [comment]
In react-native, use "npx react-native run-android" instead of "yarn android" to start project. It works for me.

**Comment by qouetos** [comment]
Hello everyone. For my part I tried lots of things... and finally I realized that I was using node 18... switching to 20 lts solved my problem.
(installation of version 20, closed all current node and java instances)
And build successfully

**Comment by bondarenkod** [comment]
1. Getting more insights: 
modify the `apps\APPNAME\android\gradlew.bat`
`"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GradleWrapperMain %* [add this >>]--info --verbose --stacktrace --scan --debug` to get more insights.

2. The error: 
`Could not move temporary workspace (D:\src\AAAAA.ReactNative_01\apps\DemoApp2\android\.gradle\8.6\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-7cac1f26-dea7-442e-a56f-947d494ffd6a) to immutable location (D:\src\AAAAA.ReactNative_01\apps\DemoApp2\android\.gradle\8.6\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a), attempting copy-then-move        
java.nio.file.AccessDeniedException: D:\src\AAAAA.ReactNative_01\apps\DemoApp2\android\.gradle\8.6\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a-7cac1f26-dea7-442e-a56f-947d494ffd6a -> D:\src\AAAAA.ReactNative_01\apps\DemoApp2\android\.gradle\8.6\dependencies-accessors\423f0288fa7dffe069445ffa4b72952b4629a15a`

- I've found that node.exe blocks a bunch of directories. 
- **D** Drive is DEV drive. 
- Windows Defender is OFF. (It always been on, turned off only for testing purposes). 
- It worked yesterday perfectly fine. What's changed over the night - have no idea.  And it has been worked for ~9 month, including the 8.6 version which I started using since rn0.74.  
- Windows' LongPath is enabled. 
- the repo was cleaned using ` git clean -xfd && git submodule foreach --recursive git clean -xfd` multiple times. 
- GRADLE_HOME is located at the D drive. The `caches` cleaned multiple times. 

![Screenshot 2024-09-21 235422](https://github.com/user-attachments/assets/db85bdea-a18f-4ff3-bcd3-67dc8abd4c03)
![Screenshot 2024-09-21 235934](https://github.com/user-attachments/assets/640e0217-f1e5-4f66-beb2-e0e60d0d739a)

Trying to understand what's going on...

**Comment by asodja** [comment]
Thanks @bondarenkod for this research. This helped us to understand one case where `Could not move temporary workspace`: version catalog dependency accessors are generated in project cache (in project .gradle folder). This can cause problem with processes that inspect project dir, e.g. NodeJS. 

In https://github.com/gradle/gradle/issues/30680 dependency accessors are generated in global cache, so should help with this issue with NodeJS a bit. This change will ship in 8.11.

**Comment by bondarenkod** [comment]
> Thanks @bondarenkod for this research. This helped us to understand one case where `Could not move temporary workspace` failure happens: version catalog dependency accessors are generated in project cache (in project .gradle folder). This can cause problem with processes that inspect project dir, e.g. NodeJS.
> 
> With #30680 dependency accessors are generated in global cache, so it should help a bit for such projects. This change will ship in 8.11.

No problem! I'm really glad my research helped you figure out the issue! 
🤝

**Comment by yulittlemoon** [comment]
> Some coments here:
> 
> * I'm not admin of my windows pc so I coudn't stop the antivirus when I run >npm run start
> * I was resolving this issue by downgrading to gradle 8.5 in the distributionUrl. Now with the latest react-native (0.75.1) I am getting the following error message:
> 
> ```
> > Failed to apply plugin 'com.android.internal.version-check'.
>    > Minimum supported Gradle version is 8.7. Current version is 8.5. If using the gradle wrapper, try editing the distributionUrl in C:\Users\c510268\Desktop\React-Native\07-componentsApp\android\gradle\wrapper\gradle-wrapper.properties to gradle-8.7-all.zip
> ```

i have same problem.have u resolved it?my node is v20.17.0

**Comment by yulittlemoon** [comment]
i have resolved this problem.
my gradle is 8.7,node is 20,and meet problem
`Immutable workspace contents have been modified" failure when workspace contents are not modified`
i try to use gradle 8.5 but get
```
> Failed to apply plugin 'com.android.internal.version-check'.
   > Minimum supported Gradle version is 8.7. Current version is 8.5. If using the gradle wrapper, try editing the distributionUrl in C:\Users\c510268\Desktop\React-Native\07-componentsApp\android\gradle\wrapper\gradle-wrapper.properties to gradle-8.7-all.zip
```
and finally i use gradle 8.7 and do
1. cd android > gradlew clean
2. npx react-native run-android
now i build successfully

**Comment by lptr** [comment]
Gradle 8.11 is out now. If you encountered this problem, can you please check if 8.11 solves it for you?

**Comment by developerdevsoni** [comment]
regrading this issue. in react native 76 version . please use the gradlew 8.11 we resolve your issue. thanks

**Comment by lptr** [comment]
We expect this has been fixed via Gradle 8.11. Please reopen if it still comes up.

**Comment by chhbzhou-boop** [comment]
兄弟萌，build.gradle.kts 文件下添加 -Didea.force.quit.on.close=true 可以解决问题， 本人版本：gradle-9.0.0，示例如下：
tasks {
    withType<JavaCompile> {
        sourceCompatibility = "21"
        targetCompatibility = "21"
        options.encoding = "UTF-8"
    }
    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
        compilerOptions.jvmTarget.set(JvmTarget.JVM_21)
        compilerOptions.freeCompilerArgs.addAll(listOf("-Xjvm-default=all"))
    }

    runIde {
        jvmArgs = listOf("-Xmx2048m", "-Didea.force.quit.on.close=true")
    }
}

**Comment by chylek-qr** [comment]
I got this error several times when using "Sync All Gradle Projects" in IntelliJ after updating to Gradle 9.1.0. No idea what it means, but I deleted the entire `/Users/chylek/.gradle/caches/9.1.0/transforms` directory, killed all Gradle daemons, and then it stopped.

**Comment by EpicVon2468** [comment]
> We expect this has been fixed via Gradle 8.11. Please comment or open a new issue if it still comes up with Gradle 8.11+.

I've got it on Gradle 9.2.1
'Caused by: java.lang.IllegalStateException: The contents of the immutable workspace '/home/mavity/.gradle/caches/9.2.1/transforms/5ea7adcdf5523c2b8c0e00feaebe17f1' have been modified. These workspace directories are not supposed to be modified once they are created. The modification might have been caused by an external process, or could be the result of disk corruption.'
<img width="1932" height="100" alt="Image" src="https://github.com/user-attachments/assets/41f5b2af-8287-4712-b434-5554585433c2" />

Strangely, nothing "bad" happens related to this.
That is to say, nothing fails because of it.
Builds still claim success, and everything _seems_ to be working just fine.
It just shows the error and floods the logs about it, but that's it.

An update: running `rm -r` on the directory claimed to be modified and then rebuilding has (at least for now?) silenced or stopped the issue. I'll come back if it happens again.

**Comment by AlexInCube** [comment]
this issue is not fixed, i used gradle 9.2.1.
i can`t run my project because of it
